vignettes/textAnnotation.Rmd

```diff
@@ -0,0 +1,230 @@
+  ),
+  list(
+    operation = "LT",
+    measure = "zoomLevel",
+    threshold = 40,
+    target = "track"
+  ))
+)
+```
+
+## Defining Visual Channels
+
+Now, lets define visual channels for track1. track1_x specifies the genomic position on the x-axis, track1_color assigns colors based on DNA bases, and track1_text specifies text annotations based on DNA bases.
+
+```{r visual_channels, eval=FALSE}
+
+track1_x <- visual_channel_x(
+  field = "position", type = "genomic"
+)
+
+track1_color <- visual_channel_color(
+  field = "base",
+  type = "nominal",
+  domain = c("A", "T", "G", "C"),
+  legend = TRUE
+)
+
+track1_text <- visual_channel_text(
+  field = "base", type = "nominal"
+)
+
+track1_style <- default_track_styles(
+  inlineLegend = TRUE
+)
+```
+
+## Creating combined track
+
+This code chunk combines the previously defined tracks (track1 and track2) into a single track (track3) and specifies various properties such as title, alignment, data, visual channels, and style.
+
+```{r combined_track, eval=FALSE}
+
+track3 <- add_single_track(
+  title = "NC_045512.2 Sequence",
+  alignment = "overlay",
+  data = track1_data,
+  tracks = add_multi_tracks(
+    track1, track2
+  ),
+  x = track1_x,
+  color = track1_color,
+  text = track1_text,
+  style = track1_style,
+  width = 800, height = 40
+)
+```
+
+## Creating the view
+
+Lets create a view (view1) that contains the combined track (track3). It specifies properties like multi-view mode, x-axis domain, alignment, and linking.
+
+```{r, compose, eval=FALSE}
+
+view1 <- compose_view(
+  multi = TRUE,
+  centerRadius = 0,
+  xDomain = list(interval = c(1, 29903)),
+  linkingId = "detail",
+  alignment = "stack",
+  tracks = add_multi_tracks(
+    track3
+  )
+)
+```
+
+## Arranging the view
+
+Next, we arrange views using the arrange_views function. It sets the title, subtitle, assembly information, layout, spacing, and includes the previously defined view1.
+
+```{r, arrange, eval=FALSE}
+
+combined_view <- arrange_views(
+  title = "SARS-CoV-2",
+  subtitle = "Data Source: WashU Virus Genome Browser, NCBI, GISAID",
+  assembly = list(list("NC_045512.2", 29903)),
+  layout = "linear",
+  spacing = 50,
+  views = list(view1),
+  listify = FALSE
+)
+```
+
+## Shiny App
+
+Finally, we define the Shiny user interface (UI) using the fluidPage function. It includes the goslingOutput function to create a placeholder for the visualization.
+We also define the Shiny server logic. It uses the renderGosling function to render the interactive visualization using the combined_view defined earlier.
+
+```{r, ui, eval=FALSE}
+
+ui <- fluidPage(
+  use_gosling(),
+  fluidRow(
+    column(6, goslingOutput("gosling_plot"))
+  )
+)
+
+
+server <- function(input, output, session) {
+  output$gosling_plot <- renderGosling({
+    gosling(
+      component_id = "sars_cov2",
+      combined_view
+    )
+  })
+}
+
+shinyApp(ui, server)
+```
+
+![](textAnnotation.png){width=500px}
+
+## Session Info
```

Comment:
I suppose that the purpose of showing the `sessionInfo` here is to provide the user reading this with the environment that has produced a working example for this vignette right?
In that sense shouldn't this one need to be hardcoded? In this case I suppose that this one will output the info about the environment parsing this rmd document right and not necesarily the environment that has produced a succesfull render of this particular shiny app
WDYT?

Reply:
I got your point. Makes sense.
But the other bioc packages like [Granges](https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html) are executing `sessionInfo()` at the end of the Vignette.
This is the intro [vignette](https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.R) for the GRanges package.